lang/ruby/lib/avro/schema.rb

```diff
@@ -33,7 +35,10 @@ class Schema
     LONG_MAX_VALUE = (1 << 63) - 1
 
     def self.parse(json_string)
-      real_parse(MultiJson.load(json_string), {})
+      json_obj = MultiJson.load(json_string)
+      schema = real_parse(json_obj, {})
+      schema.logical_type = LogicalTypes.parse(json_obj) if json_obj.is_a?(Hash)
```

Comment:
You mean pass the logical type to the inializer when instantiating the Schema object? I tried that, but it would require changing the code quite a bit – the `real_parse` method has multiple returns and separately instantiates the classes. Also, all of the Schema subclasses' `initialize` methods would have to be changed. If you're okay with the larger diff I can do it.

At this point, though, I'd prefer extracting a separate SchemaParser class to handle the parsing bits in order to keep the code cleaner. I originally didn't want to do this because I preferred the PR to be simple and focused. If you're also okay with it I can refactor a bit.